src/posthog-surveys.ts

```diff
@@ -154,27 +156,55 @@ export class PostHogSurveys {
     }
 
     loadIfEnabled() {
-        const surveysGenerator = assignableWindow?.__PosthogExtensions__?.generateSurveys
+        const phExtensions = assignableWindow?.__PosthogExtensions__
+
+        if (!phExtensions) {
+            logger.error('PostHog Extensions not found.')
+            return
+        }
+
+        const surveysGenerator = phExtensions.generateSurveys
+
+        const disableSurveys = this.instance.config.disable_surveys
+
+        if (disableSurveys) {
+            logger.info('Disabled. Not loading surveys.')
+            return
+        }
 
-        if (!this.instance.config.disable_surveys && this._decideServerResponse && !surveysGenerator) {
+        if (!this._decideServerResponse) {
+            logger.warn('Decide not loaded yet. Not loading surveys.')
+            return
+        }
+
+        if (!surveysGenerator) {
             if (this._surveyEventReceiver == null) {
                 this._surveyEventReceiver = new SurveyEventReceiver(this.instance)
             }
 
-            assignableWindow.__PosthogExtensions__?.loadExternalDependency?.(this.instance, 'surveys', (err) => {
-                if (err) {
-                    return logger.error('Could not load surveys script', err)
-                }
+            const loadExternalDependency = phExtensions.loadExternalDependency
+            if (loadExternalDependency) {
+                loadExternalDependency(this.instance, 'surveys', (err) => {
+                    if (err) {
+                        return logger.error('Could not load surveys script', err)
+                    }
 
-                this._surveyManager = assignableWindow.__PosthogExtensions__?.generateSurveys?.(this.instance)
-            })
+                    this._surveyManager = phExtensions.generateSurveys?.(this.instance)
+                })
+            } else {
+                logger.error('PostHog loadExternalDependency extension not found. Cannot load remote config.')
```

Comment:
you can bundle such that loadExternalDependency is not available but surveys is

e.g. `array-full.no-external.ts` 

i'm not completely sure how that would look to the code (it's a while since I checked)

Reply:
so for example here https://github.com/PostHog/posthog-js/blob/912e97dcbfb71dbbd20711a37f1ddc30973de712/src/extensions/replay/sessionrecording.ts#L760-L770

* we check for presence of rrweb
* if it is present then we don't have to load anything
* and we call start
* otherwise we try to load it
* and call start if that succeeds

(i bet we're not consistent about being that careful 🙈)

Reply:
Does that mean the current code already has a bug then? All I did was add conditions so I could do logging for each of the objects that are in the not-desired state.
I can fix it if that's the case.

Reply:
Done!